httomo/method_wrappers/rotation.py

```diff
@@ -68,7 +68,7 @@ def _build_kwargs(
         if not "ind" in dict_params or (
             dict_params["ind"] == "mid" or dict_params["ind"] is None
         ):
-            updated_params = {**dict_params, "ind": (dataset.global_shape[1] - 1) // 2}
+            updated_params = {**dict_params, "ind": (dataset.base.shape[1] - 1) // 2}
```

Comment:
Why was this change needed again? Isn't the index the global shape's midpoint? And given that the rotation methods are project pattern - what's the difference between the old and the new?

Reply:
Ah, possibly this doesn't need to be done now. This change was done this commit [5b0a2b0](https://github.com/DiamondLightSource/httomo/pull/209/commits/5b0a2b04d349be75245dc872f8c1fb517c93d217), which came _before_ the merge I did of `feature/transparent-file-store` into `previewing` to make use of `FullFileDataSet.data_offset` [39567b0](https://github.com/DiamondLightSource/httomo/pull/209/commits/39567b0c8c7259a5592eb6a7730fa397a3365192).

This change was needed when I was still trying to use `FullFileDataSet.global_index` to do offsets for previewing, but quite possibly it's not needed now that we're using `FullFileDataSet.data_offset`.

I'll check if this is actually needed, thanks for the keen eye!

Reply:
I get it - so I think `dataset.shape` should be enough here? 

Still, I think `FullFileDataSet.global_shape` should include the previewing / offsetting ...

Reply:
> Still, I think FullFileDataSet.global_shape should include the previewing / offsetting ...

The problem with this is that it causes this assertion to fail I think: https://github.com/DiamondLightSource/httomo/blob/dd193901bd4ae905a4df72cbbda5041af20c803c/httomo/runner/dataset.py#L44-L47

What happens is that:
- `data` is the h5py `Dataset` object
- `data.shape` is the shape of the full raw data
- whereas if we were to have `global_shape` be the previewed shape, this then is "smaller" than `data.shape` -> assertion fails

And it's a sensible assertion to be honest, so I didn't want to just remove it, but I guess we can also do that if that is an ok tradeoff to be able to have `FullFIleDataSet.global_shape` be the previewed shape?

I'm still not sure about `FullFileDataSet.global_shape` being the previewed shape, because in the context of a "full file", the global shape makes sense to be the "full shape" without any previewing, especially since `FullFileDataSet` doesn't know anything about previewing, it's purely for holding a "full file"?

It again comes back to the idea that maybe the loader shouldn't be using a `FullFileDataSet` to do its business, `FullFiIeDataSet` was originally made for the store.

Reply:
> I get it - so I think dataset.shape should be enough here?

I think this should work, using the block's shape to get the detector y/height length. In order for that to go wrong, the detector y/height dimension would have to be the slicing dim (so then the block wouldn't have the full previewed shape in the detetcor y dimension), and for the rotation wrapper we explicitly do this single sinogram gathering stuff under the assumption that it'll only ever be used in a "projection section".

Reply:
Using the block shape rather than the base shape is done [1728181](https://github.com/DiamondLightSource/httomo/pull/209/commits/17281812846a5e0b831a335330d1a2e6f5a3ea20)

Reply:
After some discussion, we decided to address the `global_shape` concerns in another branch off of `feature/transparent-file-store`... :sweat_smile: